src/main/java/org/apache/sysml/api/python/SystemML.py

```diff
@@ -37,11 +37,329 @@
 from pyspark.ml import Estimator, Model
 from scipy.sparse import spmatrix
 from scipy.sparse import coo_matrix
+from contextlib import contextmanager
+
+def simpleAddPyDMLStmt(fnName, args=None, obj=None):
+    m = Matrix(0)
+    addPyDMLStmt(fnName, lhs=m, args=args, obj=obj)
+    return m
+
+def addPyDMLStmt(fnName, lhs=None, args=None, obj=None):
+    pydmlStmt = [getTabs()]
+    if lhs is not None:
+        pydmlStmt = pydmlStmt + [str(lhs.getID()), ' = ']
+    if obj is None:
+        pydmlStmt = pydmlStmt + [ fnName, '(']
+    else:
+        pydmlStmt = pydmlStmt + [  obj, '.', fnName, '(']
+    if args is not None:
+        pydmlStmt = pydmlStmt + args
+    MLContext.globalMLContext.pydml.append(''.join(pydmlStmt + [')\n']))
+ 
+def addPyDMLString(pydmlStmt):
+    MLContext.globalMLContext.pydml.append(''.join(pydmlStmt))
+   
+# Constructor         
+def full(init, rows=None, cols=None):
+    if rows is None or cols is None:
+        raise Exception('rows and cols are mandatory arguments')
+    scopeCheck('full')
+    return simpleAddPyDMLStmt('full', args=[str(init), ', rows=', str(rows), ', cols=', str(cols)])
 
 SUPPORTED_TYPES = (np.ndarray, pd.DataFrame, spmatrix)
 
-class MLContext(object):
+def matrix(numPyArr):
+    scopeCheck('matrix')
+    if isinstance(numPyArr, np.ndarray):
+        colNames = []
+        numCols = len(numPyArr[0,:])
```

Comment:
This will blow up if a user passes in a vector, such as `np.array([1,2,3])`, which has shape `(3,)`, thus no explicit number of columns.  Instead, let's just create the Pandas DataFrame first via `pdf = pd.DataFrame(numPyArr)`, which will have a full, 2-dimensional shape such as `(3,1)`.  Then, we can get the number of columns with `pdf.shape[1]`.  The column names list can be created using a Pythonic for-comprehension via `columns = ["C" + str(i) for i in range(pdf.shape[1])]`.  (The `0` in `range(0, numCols)` is not necessary).  Then, the Pandas DataFrame columns can be updated via `pdf.columns = columns`.